main/src/main/scala/org/clulab/processors/Document.scala

```diff
@@ -78,3 +113,108 @@ object Document {
     Document(doc.id, doc.sentences, doc.coreferenceChains, doc.discourseTree, doc.text)
 
 }
+
+/**
+  * This is the interface required for interaction with Document
+  * Since the Document is serializable, everything in it should be, including the DocumentAttachments.
+  */
+trait DocumentAble extends Serializable
+
+/**
+  * This is the interface used to build DocumentAttachments from text when
+  * they are deserialized by the DocumentSerializer as coded further below.
+  */
+trait DocumentAttachmentBuilderFromText {
+  def mkDocumentAttachment(text: String): DocumentAttachment
+}
+
+/**
+  * This is an implementation of DocumentAttachmentBuilderFromText that indeed
+  * creates an object from text, but it is an illegible text version of the Java
+  * object serialization.  It is used as a backup absent a more legible representation.
+  * Subclasses must have an empty constructor which can be called using reflection
+  * based on the name in DocumentAttachment.documentAttachmentBuilderClassName by
+  * the DocumentSerializer.  The constructed builder is then used to construct the
+  * DocumentAttachment by calling mkDocumentAttachment which likely returns a
+  * customized subclass of DocumentAttachment.
+  */
+class ObjectDocumentAttachmentBuilderFromText extends DocumentAttachmentBuilderFromText {
+
+  /**
+    * This matches the format in DocumentAttachment.toDocumentSerializer.
+    */
+  def mkDocumentAttachment(text: String): DocumentAttachment = {
+    val byteArray: Array[Byte] = text.split(',').map { value =>
+      val byte: Byte = java.lang.Byte.decode(value)
+      byte
+    }
```

Comment:
I believe these 4 lines can be written as `val byteArray = text.split(",").map(java.lang.Byte.decode)`, and similarly when these lines of code are repeated below.

Reply:
Thanks, I'll check.  Scala was doing too many or too few conversions and this *seemed* to be the way to avoid it.